src/tests/server/database/user.test.js

```diff
@@ -1,87 +1,69 @@
-import test, { after, before, describe } from "node:test";
+import test, { after, afterEach, before, describe } from "node:test";
 import assert from "node:assert/strict";
 import * as User from "../../../server/database/user.js";
 import {
   initTestDatabase,
   concludeTesting,
-  Models,
+  clearTestData,
 } from "../../../server/database/index.js";
+import { createAdminUser, createUser } from "../../test-helpers.js";
 
 describe(`user tests`, async () => {
   before(async () => await initTestDatabase());
+  afterEach(() => clearTestData());
   after(() => concludeTesting());
```

Comment:
Because we're importing `test-helpers`, which in turn has an `import { stdin } from "../setup/utils.js";` we also need to make sure to close that stdio again or we're going to be waiting forever for testing to finish =)

```suggestion
import { createAdminUser, createUser } from "../../test-helpers.js";
import { closeReader } from "../../../setup/utils.js";

describe(`user tests`, async () => {
  before(async () => await initTestDatabase());
  afterEach(() => clearTestData());
  after(() => {
    concludeTesting();
    closeReader();
  });
```

You can confirm this by running `npm test`, which will look like it's getting stuck on the project tests, but actually gets stuck on the next file.